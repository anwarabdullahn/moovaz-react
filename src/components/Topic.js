import React from 'react';
import { Card, CardText, CardBody, Row, Col } from 'reactstrap';

const Topic = (props) => {
	console.log('props', props);

	const addTopic = (
		<Col className="my-3" lg={3} md={3} sm={6} onClick={props.addHandle}>
			<Card
				style={{
					borderStyle: 'dashed',
					height: '210px',
					backgroundColor: '#cecece',
					color: '#fff',
					borderWidth: '7px'
				}}
			>
				<CardBody className="d-flex justify-content-center align-items-center">
					<h3>
						Add New Topic <i className="fas fa-plus" />
					</h3>
				</CardBody>
			</Card>
		</Col>
	);

	const topicList = props.topics.map((topic) => (
		<Col className="my-3" lg={3} md={3} sm={6} key={topic.id}>
			<Card style={{ height: '210px' }}>
				<CardBody>
					<h4 className="text-center">{topic.title}</h4>
					<CardText className="d-flex mt-5">
						<span
							className="ml-auto"
							style={{ position: 'relative', right: '50px' }}
							onClick={() => props.upVoteHandle(topic.id)}
						>
							<i className="fas fa-2x fa-thumbs-up" style={{ color: 'green' }} /> {topic.upvote}
						</span>
						<span className="mr-auto" style={{ position: 'relative', left: '50px' }}>
							<i className="fas fa-2x fa-thumbs-down" style={{ color: 'red' }} /> {topic.downvote}
						</span>
					</CardText>
				</CardBody>
			</Card>
		</Col>
	));

	return (
		<Row>
			{topicList}
			{addTopic}
		</Row>
	);
};

export default Topic;
