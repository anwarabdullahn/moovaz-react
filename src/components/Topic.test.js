import React from 'react';
import { shallow } from 'enzyme';
import Topic from './Topic';
describe('Topic', () => {
	it('should render topic correctly in "debug" mode', () => {
        expect(shallow(<Topic debug />)).toMatchSnapshot();
	});
});
