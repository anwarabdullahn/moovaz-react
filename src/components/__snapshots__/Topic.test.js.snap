// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topic should render topic correctly in "debug" mode 1`] = `
<Row
  tag="div"
>
  <Col
    className="my-3"
    lg={3}
    md={3}
    sm={6}
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <Card
      style={
        Object {
          "backgroundColor": "#cecece",
          "borderStyle": "dashed",
          "borderWidth": "7px",
          "color": "#fff",
          "height": "210px",
        }
      }
      tag="div"
    >
      <CardBody
        className="d-flex justify-content-center align-items-center"
        tag="div"
      >
        <h3>
          Add New Topic 
          <i
            className="fas fa-plus"
          />
        </h3>
      </CardBody>
    </Card>
  </Col>
</Row>
`;
